src/Microsoft.Azure.WebJobs.Host/Listeners/ListenerFactoryContext.cs

```diff
@@ -18,7 +18,35 @@ public class ListenerFactoryContext
         private readonly SharedQueueHandler _sharedQueue;
         private IDispatchQueueHandler _dispatchQueue;
 
+        /// <summary>
+        /// Creates a new instance
+        /// </summary>
+        /// <param name="descriptor">The <see cref="FunctionDescriptor"/> to create a listener for.</param>
+        /// <param name="executor">The <see cref="ITriggeredFunctionExecutor"/> that should be used to invoke the
+        /// target job function when the trigger fires.</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> to use.</param>
+        public ListenerFactoryContext(FunctionDescriptor descriptor, ITriggeredFunctionExecutor executor, CancellationToken cancellationToken)
```

Comment:
Adding this public ctor back - it was removed when this work was done, which was a breaking change. The shared queue capability is preserved via an internal constructor.